### DISCUSSION TITLE
Misunderstanding regarding namespaces

### DISCUSSION THREAD

**Discussion by Shahor** [socketio/socket.io#4996] [Q&A]
Hi, 

I'm trying to wrap my head around namespaces but I'm really confused at the moment. 

If I get the documentation correctly, namespaces should automatically use multiplexing on the client and re-use available socket connections (provided the origin is the same) when re-establishing new connections. 

This sounds awesome and is exactly what I want to implement, but I'm unable to do so. I was afraid the setup of my sockets in the app I'm working on was too complex so I reduced the code to its bare minimum for a reproduction and it seems I'm having the same issue. 

The repro repo can be found here: https://github.com/Shahor/socketio-namespace-example

This is a monorepo using nestjs for the backend and nextjs for the frontend, powered by nx as a monorepo but hopefully that doesn't impact our use case.

As you can see if you start the repo and navigate using the links, you can see in the network tab that each page re-triggers a new websocket connection instead of re-using the one already available from the wrapping `_app` component. 

What am I not understanding? This seems to be the exact use case for multiplexing and if I'm not mistaken NextJS behaves as an SPA here so I'm not sure what is happening.

**Comment by jokester** [comment]
After a look at the [code](https://github.com/socketio/socket.io-client/blob/main/lib/index.ts) I think this is how transport multiplex works:

Internally `io(url)` converts url to an `id` and reuses `Manager` (and transports within) across same ids. The id contains url path too, so to reuse transport multiple url-s have to at least contain the same path. Maybe they have to be identical.

**Reply by Shahor** [reply]
Thank you @jokester, I didn't have time to go back to this until now but I'll be investigating again soon and will let you know. 

🙇🏽
